src/WebJobs.Extensions.DurableTask/AzureStorageAccountProvider.cs

```diff
@@ -19,6 +21,9 @@ internal sealed class AzureStorageAccountProvider : IStorageAccountProvider
 #if !FUNCTIONS_V1
         private readonly ITokenCredentialFactory credentialFactory;
 
+        private readonly ConcurrentDictionary<string, TokenCredential> cachedTokenCredentials =
+            new ConcurrentDictionary<string, TokenCredential>();
+
```

Comment:
Is there a situation where we should remove elements from this dictionary?

Reply:
I can't think of one. I believe we want all of these for the lifetime of the application.